expr/src/Error.hs

```diff
@@ -0,0 +1,19 @@
+module Error(Error(..)) where
+
+import qualified Expr
+
+data Error
+  = DivByZero Expr.Expr Expr.Expr
+  | NegativeSqrtNum Expr.Expr
+  | UnassignedVar String
+
+instance Show Error where
+  show (DivByZero a b) = "Division by zero error: " ++ show a ++ " / " ++ show b
+  show (NegativeSqrtNum s) = "Cannot take square root of negative number: " ++ show s
+  show (UnassignedVar s) = "Unassigned variables: " ++ s
+
+instance Eq Error where
+  DivByZero _ _ == DivByZero _ _ = True
+  NegativeSqrtNum _ == NegativeSqrtNum _ = True
+  UnassignedVar _ == UnassignedVar _ = True
+  _ == _ = False
```

Comment:
It's better to include expressions in the comparison of the errors. And as in any other case, it's better to derive whenever your implementation is standard.

Reply:
Fixed